### DISCUSSION TITLE
run both TCP master and TCP slave at same time?

### DISCUSSION THREAD

**Discussion by test2012vnz** [espressif/esp-modbus#59] [Q&A] [answered]
Hi,
is there any suggestion to run both TCP master and TCP slave modbus at same time at same ?

**Comment by alisitsyn** [comment, marked as answer]
Hi @test2012vnz ,

Thank you for the question. The stack esp-modbus v1.0.x supports one Master and one Slave of each type. 
So, yes the TCP master and TCP slave can work. The [esp-modbus v2.0](https://github.com/espressif/esp-modbus/discussions/86) supports several instances of Modbus Master and Slave with its independent setup options and possible handling in the same application. The examples describe creation of one instance at a time as it was in previous stack. The code of Modbus Master and Slaves can be combined in your application to realize complex functionality. Also, you can take a look to [test_apps](https://github.com/espressif/esp-modbus/blob/main/test_apps/physical_tests/main/test_modbus_tcp_comm_master_slave.c#L238) to get more ideas. Please describe your project and idea to let me identify possibility and limitation in this case and provide suggestions.
